From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sheetal <sheetal@nvidia.com>
Date: Wed, 25 Mar 2026 10:14:24 +0000
Subject: [PATCH 196/220] ASoC: tegra: Use dev_err_probe() in tegra186_asrc
 probe

Log errors in the Tegra186 ASRC probe path using dev_err_probe().

Signed-off-by: Sheetal <sheetal@nvidia.com>
Link: https://patch.msgid.link/20260325101437.3059693-2-sheetal@nvidia.com
Signed-off-by: Mark Brown <broonie@kernel.org>
(cherry picked from commit 884f3101d1edb2bff463a1a4bae162ab011fe73f)

diff --git a/sound/soc/tegra/tegra186_asrc.c b/sound/soc/tegra/tegra186_asrc.c
index 2e0ffb75e052..6ce7ddb788a5 100644
--- a/sound/soc/tegra/tegra186_asrc.c
+++ b/sound/soc/tegra/tegra186_asrc.c
@@ -1016,10 +1016,9 @@ static int tegra186_asrc_platform_probe(struct platform_device *pdev)
 	err = devm_snd_soc_register_component(dev, &tegra186_asrc_cmpnt,
 					      tegra186_asrc_dais,
 					      ARRAY_SIZE(tegra186_asrc_dais));
-	if (err) {
-		dev_err(dev, "can't register ASRC component, err: %d\n", err);
-		return err;
-	}
+	if (err)
+		return dev_err_probe(dev, err,
+				     "can't register ASRC component\n");
 
 	pm_runtime_enable(dev);
 
-- 
2.43.0

